services/src/model_training/ml_tasks.rs

```diff
@@ -0,0 +1,177 @@
+use crate::contexts::Context;
+use crate::datasets::upload::UploadId;
+use crate::error;
+use crate::handlers::model_training::{
+    accumulate_raster_data, get_operators_from_workflows, get_query_processors,
+};
+use crate::tasks::{Task, TaskId, TaskManager, TaskStatusInfo};
+use crate::workflows::workflow::Workflow;
+use geoengine_datatypes::error::ErrorSource;
+use geoengine_datatypes::primitives::MachineLearningQueryRectangle;
+use geoengine_datatypes::util::Identifier;
+use geoengine_operators::engine::ExecutionContext;
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+use std::path::PathBuf;
+use std::sync::Arc;
+use tokio::fs;
+use utoipa::ToSchema;
+
+use super::{xgb_train_model, XgboostTrainingParams};
+
+use serde_json::Value;
+
+/// By default, we set [`RasterDatasetFromWorkflow::as_cog`] to true to produce cloud-optmized `GeoTiff`s.
+#[inline]
+const fn default_as_cog() -> bool {
+    true
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, ToSchema)]
+#[schema(example = json!({"name": "foo", "description": null, "query": {"spatialBounds": {"upperLeftCoordinate": {"x": -10.0, "y": 80.0}, "lowerRightCoordinate": {"x": 50.0, "y": 20.0}}, "timeInterval": {"start": 1_388_534_400_000_i64, "end": 1_388_534_401_000_i64}, "spatialResolution": {"x": 0.1, "y": 0.1}}}))]
+pub struct MLTrainRequest {
+    pub params: XgboostTrainingParams,
+    pub input_workflows: Vec<Workflow>,
+    pub label_workflow: Vec<Workflow>,
+    pub query: MachineLearningQueryRectangle,
+}
+
+/// parameter for the dataset from workflow handler (body)
+#[derive(Clone, Debug, Deserialize, Serialize, ToSchema)]
+#[schema(example = json!({"name": "foo", "description": null, "query": {"spatialBounds": {"upperLeftCoordinate": {"x": -10.0, "y": 80.0}, "lowerRightCoordinate": {"x": 50.0, "y": 20.0}}, "timeInterval": {"start": 1_388_534_400_000_i64, "end": 1_388_534_401_000_i64}, "spatialResolution": {"x": 0.1, "y": 0.1}}}))]
+pub struct MachineLearningModelFromWorkflow {
+    pub name: String,
+    pub description: Option<String>,
+    pub query: MachineLearningQueryRectangle,
+
+    // TODO: is that cog stuff relevant in this case?
+    #[schema(default = default_as_cog)]
+    #[serde(default = "default_as_cog")]
+    pub as_cog: bool,
+}
+
+/// response of the machine learning model from workflow handler
+#[cfg(feature = "xgboost")]
+#[derive(Clone, Debug, Deserialize, Serialize, ToSchema)]
+pub struct MachineLearningModelFromWorkflowResult {
```

Comment:
set `#[serde(rename_all = "camelCase")]`  for all structs that are part of the API